### PR TITLE
Javascript capstone project

### PR DESCRIPTION
**Pull request by Bornittah**
### Project requirements
**APIs**
Use [Football Standings API](https://github.com/azharimm/football-standings-api) to:

- Get a list of items with a unique item id (or generate the unique id).
- For a given item, get detailed information about it.

Use our [Involvement API](https://www.notion.so/microverse/Involvement-API-869e60b5ad104603aa6db59e08150270) to record the different user interactions (likes and comments )

**Interfaces**
Build these interfaces:

- The home page.
- The comments popup.

**Home page**
When the page loads, the webapp retrieves data from:

- The selected API and shows the list of items on screen.
- The Involvement API to show the item likes.

The page makes only 2 requests:

- One to the base API.
- And one to the Involvement API.

**Interactions**

- When the user clicks on the Like button of an item, the interaction is recorded in the Involvement API and the screen is updated.
- When the user clicks on the "Comments" button, the Comments popup appears.
- When the popup loads, the webapp retrieves data from:
- The selected API and shows details about the selected item.
- The Involvement API to show the item comments.
- When the user clicks on the "Comment" button, the data is recorded in the Involvement API and the screen is updated.
**Counters** 
We have counters in all the interfaces that show:
- The number of items (home).
- The number of comments (comments popup).
- These count functions need to be covered with unit tests using Jest.

**Technical set up**

- Set up the repository on GitHub and use Gitflow.
- Set up webpack.
- Set up a JavaScript testing library (Jest).

### CODE REVIEW

File: src/modules/user-interactions.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    list.forEach((data) => {
+      str += ` <li class="flex">
+      <p>${data.creation_date}</p>
+      <p><b>${data.username}:</b></p>
+      <p>${data.comment}</p>
+    </li>`;
+    });
+    document.querySelector('.comments').innerHTML = str;
+  }
+};
```
Comment (AnselemOdims) [inline src/modules/user-interactions.js:62]: - [ ] Good job working on this, but the comments on the modal popup does not get fetched when the modal opens not until a new comment is made. Please kindly ensure that the comments get loaded and displayed when the modal is popped up

File: src/modules/app.test.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  </p>
+</footer>`;
+
+  test('Number of Comments', () => {
+    const ul = document.querySelectorAll('.comments');
+    ul.innerHTML = '<li>2022/06/29 Aga: good game</li>';
+    const counter = document.querySelectorAll('.comments_count');
+    const comments = ul.childElementCount;
+    expect(counter.textContext).toBe(comments);
+  });
```
Comment (AnselemOdims) [inline src/modules/app.test.js]: - [ ] You are expected to create a function to calculate counter numbers on each page(Home and comments). This function is what you are going to write tests for. Please kindly create the function and write tests for it.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @Bornittah , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please record a presentation video of no more than 5mins in length explaining the features of the project and highlighting a piece of code you are most proud of.
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by KDlamini — APPROVED** [review]
### Hi @Bornittah 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

## Highlights 🌞 
✔️  Good PR title and description
✔️  Good UI and wireframe parsing
✔️  API integration implemented properly
✔️  ES6 and modules are used correctly
✔️  Commit messages are meaningful
✔️  No linter errors

# Status: Approved ✔️ 

giphyapproved
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**